File: javascript/sample6.test.js
import Plateau from "./plateau";
import MarsRover from "./mars-rover";

describe('Mars Rover', function() {

    it('can follow instructions', function () {
        var plateau = new Plateau(5, 5);
        var rover = new MarsRover(plateau, [1, 2], "N");

        var forward = Math.random()*10
        for (let i = 0; i < forward; i++) {
            rover.forward();
        }

        if (forward < 5) {
            expect(rover.status).toEqual("OK");
        } else {
            expect(rover.status).toEqual("NOK");
        }

    });

});

File: go/marsroverkata/sample6_test.go
package marsrover

import (
	"github.com/stretchr/testify/assert"
	"math/rand"
	"testing"
)

func TestReceiveCommands(t *testing.T) {
	plateau := Plateau{maxX: 5, maxY: 5}
	startingPosition := Coordinates{1,2}
	marsRover := MarsRover{plateau: plateau, heading: N, position: startingPosition}

	forward := rand.Intn(10)
	for i := 0; i < forward; i++ {
		marsRover.forward()
	}

	if forward < 5 {
		assert.Equal(t, marsRover.status, OK)
	} else {
		assert.Equal(t, marsRover.status, NOK)
	}
}
